src/registrar/models/domain.py

```diff
@@ -1666,6 +1670,51 @@ def _update_hosts_and_contacts(self, cleaned, fetch_hosts, fetch_contacts):
             if old_cache_contacts is not None:
                 cleaned["contacts"] = old_cache_contacts
 
+    def _update_hosts_and_ips_in_db(self, cleaned, fetch_hosts):
+        """Update hosts and host_ips in database if retrieved from registry.
+
+        Parameters:
+            self: the domain to be updated with hosts and ips from cleaned
+            cleaned: dict containing hosts.  Hosts are provided as a list of dicts, e.g.
+                [{"name": "ns1.example.com",}, {"name": "ns1.example.gov"}, "addrs": ["0.0.0.0"])]
+            fetch_hosts: boolean indicating whether or not fetch_hosts was called
+        """
+        if fetch_hosts:
+            cleaned_hosts = cleaned["hosts"]
+            # Get all existing hosts from the database for this domain
+            existing_hosts_in_db = Host.objects.filter(domain=self)
+            # Identify hosts to delete
+            cleaned_host_names = set(cleaned_host["name"] for cleaned_host in cleaned_hosts)
```

Comment:
**Question:** Would you mind giving an example of when we would have duplicates of a host name? (Just trying to think through all edge cases)

Reply:
There should not be a case of duplicates.  It is not allowed by the logic in the name server form; throws an error 'Remove duplicate entry'.  And it is not allowed by the registry.